src/main/java/hudson/remoting/Channel.java

```diff
@@ -832,9 +834,15 @@ public void setJarCache(@Nonnull JarCache jarCache) {
      */
     public <V,T extends Throwable>
     V call(Callable<V,T> callable) throws IOException, T, InterruptedException {
+        return call(callable, UserRequest.DEFAULT_PERFORM_TIMEOUT, UserRequest.DEFAULT_EXECUTION_TIMEOUT);
+    }
+    
+    public <V,T extends Throwable>
+    V call(@Nonnull Callable<V,T> callable, @CheckForNull Duration performTimeout, @CheckForNull Duration executionTimeout) 
+            throws IOException, T, InterruptedException {
```

Comment:
I interpreted the TODO in the PR as referring to methods added to `VirtualChannel` with `default` implementations. This is just being added to a `class` so there is no Java 8 dependency. You chose the `throws` clause for the method and I am arguing that it should be throwing `TimeoutException`. See signatures of `Future.get` and so on.